perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -61,53 +236,72 @@ def GetInfo():
-     attached disk is large enough to support the fio benchmark.
+     This includes installing fio, bc, and libaio1 and insuring that
+     the attached disk is large enough to support the fio
+     benchmark. We also make sure the job file is always located at
+     the same path on the local machine.
 
   Args:
     benchmark_spec: The benchmark specification. Contains all data that is
         required to run the benchmark.
+
   """
-  vms = benchmark_spec.vms
-  vm = vms[0]
+
+  if FLAGS.fio_jobfile:
+    ignored_flags = {'--' + flag_name
+                     for flag_name in FLAGS_IGNORED_FOR_CUSTOM_JOBFILE
+                     if FLAGS[flag_name].present}
+
+    if ignored_flags:
+      logging.warning('Fio job file specified. Ignoring options "%s"',
+                      ', '.join(ignored_flags))
+  if FLAGS['device_fill_size'].present and not FLAGS['against_device'].present:
+    logging.warning('--device_fill_size has no effect without --against_device')
+
+  vm = benchmark_spec.vms[0]
   logging.info('FIO prepare on %s', vm)
   vm.Install('fio')
-  file_path = data.ResourcePath(flags.FLAGS.fio_jobfile)
-
-  disk_size_kb = vm.GetDeviceSizeFromPath(vm.GetScratchDir())
-  amount_memory_kb = vm.total_memory_kb
-  if disk_size_kb < amount_memory_kb * flags.FLAGS.memory_multiple:
-    logging.error('%s must be larger than %dx memory"',
-                  vm.GetScratchDir(),
-                  flags.FLAGS.memory_multiple)
-    # TODO(user): exiting here is probably not the correct behavor.
-    #    When FIO is run across a data set which is too not considerably
-    #    larger than the amount of memory then the benchmark results will be
-    #    invalid. Once the benchmark results are returned from Run() an
-    #    invalid (or is that rather a 'valid' flag should be added.
-    exit(1)
+
+  # Fill the disk or file we're using
+  disk = vm.scratch_disks[0]
   if FLAGS.against_device:
-    device_path = vm.scratch_disks[0].GetDevicePath()
-    logging.info('Umount scratch disk on %s at %s', vm, device_path)
-    vm.RemoteCommand('sudo umount %s' % vm.GetScratchDir())
-    logging.info('Fill scratch disk on %s at %s', vm, device_path)
+    mount_point = disk.mount_point
+    logging.info('Umount scratch disk on %s at %s', vm, mount_point)
+    vm.RemoteCommand('sudo umount %s' % mount_point)
+
+    if FLAGS.device_fill_size:
+      device_path = disk.GetDevicePath()
+      logging.info('Fill scratch disk on %s at %s', vm, device_path)
+      command = (
+          ('sudo %s --filename=%s --ioengine=libaio '
+           '--name=fill-device --blocksize=512k --iodepth=64 '
+           '--rw=write --direct=1 --size=%s') %
+          (fio.FIO_PATH, device_path, FLAGS.device_fill_size))
+      vm.RemoteCommand(command)
+  else:
+    file_path = posixpath.join(disk.mount_point, DEFAULT_TEMP_FILE_NAME)
+    # We compute in MB to avoid rounding errors with 1GB disks.
+    size = str(
```

Comment:
This size only applies to the prewarming, not the job file. I will make that more clear.

The regular size parameter doesn't apply to the generated files, because they use time-based runs, but I do plan to add a flag for working set size. You can expect that patch very soon.